draft-ietf-protected-audience-key-value-services.md

```diff
@@ -318,9 +318,43 @@ Each key group is expected to have exactly one tag from the following list:
 | renderUrls | "renderUrls" represent URLs for advertisements to be looked up from the service's internal datastore. |
 | adComponentRenderUrls | "adComponentRenderUrls" represent component URLs for advertisements to be looked up from the service's internal datastore. |
 
-### Generating a Request
-
-TODO
+### Generating a Request {#request-generate}
+
+This section describes how the client MAY form and serialize request messages in order to fetch values from the Trusted Key Value server. 
+
+This algorithm takes as input an [HPKE] `public key` and its associated `key id`, a `metadata` map for global configuration, where both keys and values are strings, and a list of `signals`, each of which is a map, from a list of strings to a list of strings.
+
+The output is an [HPKE] ciphertext encrypted `request` and a context `request context`.
+
+1. Let `request map` be an empty map.
+1. Let `compression group id` be 0.
+1. Let `partitions` be an empty array.
+1. For each `signal` in `signals`:
+    1. Let `partition` be an empty map.
+    1. Set `partition["id"]` to 0.
```

Comment:
Should we be setting `partition["id"]` to `compression group id` here? Otherwise all the partitions will have an ID of 0.